### PR TITLE
Implement full model finetuning

### PR DESCRIPTION
**Pull request by OrianeN**
Implemented via a new config parameter `load_pretrained_model` and a new method `load_state_dict_from_pretrained()` in class `SimpleModel`.

Enable finetuning of another PaPie model by loading its state dict into the current model.

Customization of which parts to load is possible via the subparameter `load_pretrained_model["exclude"]`.

I already used my solution in an experiment in which I finetuned a PaPie POS tagger for Occitan, pretrained on a large synthetic dataset, finetuned with a smaller manually annotated dataset.

The results for the pretrained POS tagger are 91,19 / 82,72 / 89,24 (all tokens, unknown tokens, ambiguous tokens) ; the results of the finetuned POS tagger (tried only with one config, best of 5 runs): 92,64 / 86,14 / 91,02.

I can also confirm that the state_dicts were successfully loaded, as I can see in the logs:
```
Loading pretrained model
Initialized 106/111 char embs
Initialized 2748/9818 fwd LM word parameters
Initialized 2748/9818 bwd LM word parameters
```
Not all parameters of the LM layers could be updated because the vocabulary size has changed (20002 for the pretrained model vs. 9818 for the finetuned model).

(I also developped it in a notebook to observe the state_dict tensors in all loading steps)

By the way I noticed that the `load_pretrained_encoder` param might not be working, as I'm not sure that `pie.Encoder.load()` method can be called like this (`Encoder` doesn't seem to be imported in the `pie.__init__`), and this method calls `pie.dataset.MultiLabelEncoder`, yet `MultiLabelEncoder` seems to have moved to `pie.data.dataset.MultiLabelEncoder`.
Since my solution enables loading only the encoder of a model, should I try to change the code so that the parameter `load_pretrained_encoder` also points to the `load_state_dict_from_pretrained()` with `exclude=["lm", *tasks_names*]` ?

### CODE REVIEW

**Comment by OrianeN** [comment]
New changes since April include some bug fixes, minor improvements such as prints/logs during the initialization phase, and a new option value "replace_fill" for config option "load_pretrained_model"/"labels_mode" (that enables replacing the parent vocab with a new one computed from the fine-tuning data + completed with some parent labels if there is space left).

I believe this PR is now ready for review.

**Comment by OrianeN** [comment]
I just added a section to the README to explain how to finetune PaPie models.
Don't hesitate if you'd still prefer to have this section in a separated file, and/or if you'd like to have other details in it.

File: README.md [hunk 1]
```diff
@@ -33,12 +33,15 @@ If you find `pie` useful, please use the following reference:
 
 ## 1. Installation
 
-PIE is available from pypi, which means that all you should need to do is:
+PIE is available from pypi (`pip install nlp-pie`), yet this fork has no published package on PyPI, so you should do:
```
Comment (PonteIneptique) [inline README.md]: It is installable through https://pypi.org/project/PaPie/

File: pie/scripts/train.py [hunk 2]
```diff
@@ -192,15 +269,8 @@ def run(settings):
...
     parser = argparse.ArgumentParser()
     parser.add_argument('config_path', nargs='?', default='config.json')
-    parser.add_argument('--opt_path', help='Path to optimization file (see opt.json)')
-    parser.add_argument('--n_iter', type=int, default=20)
+    parser.add_argument("--seed", type=int, default=None)
     args = parser.parse_args()
 
     settings = settings_from_file(args.config_path)
-
-    from pie import optimize
```
Comment (PonteIneptique) [inline pie/scripts/train.py:201]: Is there a reason for this deletion ? Can't remember ?

File: README.md (hunk 1 above)
Comment (OrianeN) [reply README.md]: Oh I didn't know about this package, I'm going to indicate it in the README then !

File: pie/scripts/train.py (hunk 2 above)
Comment (OrianeN) [reply pie/scripts/train.py:201]: From what I recall, I was getting a ModuleNotFound error when running the script in debug mode, but maybe it works in a normal setup. I'm going to rollback that change, thanks for noticing.
